core/modules/filter-tracker.js

```diff
@@ -0,0 +1,108 @@
+/*\
+title: $:/core/modules/filter-tracker.js
+type: application/javascript
+module-type: global
+
+Class to track the results of a filter string
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+function FilterTracker(wiki) {
+	this.wiki = wiki;
+	this.trackers = [];
+	this.nextTrackerId = 1;
+}
+
+FilterTracker.prototype.handleChangeEvent = function(changes) {
+	this.processTrackers();
+	this.processChanges(changes);
+};
+
+/*
+Add a tracker to the filter tracker. Returns null if any of the parameters are invalid, or a tracker id if the tracker was added successfully. Options include:
```

Comment:
This method currently always returns a tracker id regardless of the validity of the parameters.